### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by azazellochg**
Related to https://github.com/scipion-em/scipion-pyworkflow/issues/476

- izip from python 2 is removed
- regex expressions fixed
- pkg_resources.parse_version replaced by packaging.version
- update requirements

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 scipion-pyworkflow>=3.0.15
-biopython==1.79
-tifffile<=2021.11.2
+biopython==1.80
+tifffile<=2023.1.23.1
 emtable
-scipy<=1.8.1 # Fix for dependency conflict with PyWavelets --> scipy 1.9 --> numpy > pyworkflow numpy
+scipy<=1.10.0
```
Comment (pconesa) [inline requirements.txt:5]: Why removing this: scipy<=1.8.1 # Fix for dependency conflict with PyWavelets --> scipy 1.9 --> numpy > pyworkflow numpy

**Comment by azazellochg** [comment]
Hi, because pywavelets (xmipp) installs okay with new numpy in pw and new
scipy (pwem). The conflict was because of ancient numpy 1.18.1 that we
still use
Best regards,
Grigory

--------------------------------------------------------------------------------
Grigory Sharov, Ph.D.

MRC Laboratory of Molecular Biology,
Francis Crick Avenue,
Cambridge Biomedical Campus,
Cambridge CB2 0QH, UK.
tel. +44 (0) 1223 267228 <+44%201223%20267228>
e-mail: ***@***.***

File: requirements.txt (hunk 1 above)
Comment (azazellochg) [reply requirements.txt:5]: @pconesa xmipp supports up to 1.10.0: https://github.com/I2PC/scipion-em-xmipp/blob/devel/requirements.txt#L4C1-L4C14

**Comment by azazellochg** [comment]
I'd like to check version parsing before merging

**Comment by azazellochg** [comment]
> I'd like to check version parsing before merging

Done!
@pconesa I'm happy to discuss the PR :)

**Review by pconesa — APPROVED** [review]

**Comment by azazellochg** [comment]
Let me test tifffile and biopython for a week or so, to double check that we don't have things like wrong tif dimensions etc.

**Comment by azazellochg** [comment]
All good on my side!

**Pull request merged**